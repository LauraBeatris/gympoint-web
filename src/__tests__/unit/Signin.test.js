import React from 'react';
import { useDispatch } from 'react-redux';
import { shallow, mount } from 'enzyme';

import Signin from '~/pages/SignIn';

jest.mock('react-redux');

describe('Signin Page', () => {
  function getWrapper() {
    return shallow(<Signin />);
  }

  it('should submit successfully', () => {
    const wrapper = mount(<Signin />);
    const form = wrapper.find('Form');

    const dispatch = jest.fn();

    useDispatch.mockImplementation(dispatch);

    form.prop('onSubmit')({ email: 'admin@gympoint.com', password: '123456' });

    expect(wrapper).toMatchSnapshot();
    expect(useDispatch).toHaveBeenCalledTimes(1);
  });
});
