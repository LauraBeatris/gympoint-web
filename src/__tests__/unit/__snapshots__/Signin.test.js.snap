// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin Page should submit successfully 1`] = `ShallowWrapper {}`;
